src/erlcloud_sns.erl

```diff
@@ -322,9 +324,9 @@ list_topics_all(Config) ->
     list_all(fun list_topics/2, topics, Config, undefined, []).
 
 
--spec list_subscriptions_by_topic/1 :: (string()) -> proplist().
--spec list_subscriptions_by_topic/2 :: (string(), string() | aws_config()) -> proplist().
--spec list_subscriptions_by_topic/3 :: (string(), undefined | string(), aws_config()) -> proplist().
+-spec list_subscriptions_by_topic/1 :: (parameter()) -> proplist().
+-spec list_subscriptions_by_topic/2 :: (parameter(), parameter() | aws_config()) -> proplist().
```

Comment:
Line 328 is too long: -spec list_subscriptions_by_topic/2 :: (parameter(), parameter() | aws_config()) -> proplist()..